cmd/exo/cmd/request.go

```diff
@@ -2,10 +2,114 @@ package cmd
 
 import (
 	"fmt"
+	"math/rand"
+	"sync"
+	"time"
 
 	"github.com/exoscale/egoscale"
+	"github.com/vbauerster/mpb"
+	"github.com/vbauerster/mpb/decor"
 )
 
+type task struct {
+	egoscale.AsyncCommand
+	string
+}
+
+// asyncTasks message variable must have same size with cmds
+func asyncTasks(tasks []task) ([]interface{}, []error) {
+
+	//init results
+	errors := &[]error{}
+	r := make([]interface{}, len(tasks))
+	responses := &r
```

Comment:
would having

```
errors := make([]error, 0, len(tasks))
responses := make([]interface{}, 0, len(tasks))
```

help in avoiding having pointers to them all around?

Reply:
for errors and responses I must have pointers !

Reply:
L37 `go execTask(task, i, c, responses, errors)`

Reply:
the reason for the pointer is the `append` calls, not the `go` calls.

Reply:
yes but with go i can't return it so i give it into parameters

Reply:
calling `execTask(..., responses, errors)` doesn't create a copy of the two slices. `append` might do.

Reply:
I dont want to copy them !

Reply:
```Golang
errors := &[]error{}
r := make([]interface{}, len(tasks))
```
and i dont put size of `errors` because if no errors it have to be empty

Reply:
you dont like pointers

Reply:
```golang
es := make([]error, 0, len(tasks))
len(es) == 0
```

Reply:
👍 your are right